exporter/exporterhelper/internal/bounded_memory_queue.go

```diff
@@ -37,17 +37,22 @@ func (q *boundedMemoryQueue[T]) Offer(ctx context.Context, req T) error {
 	}
 
 	select {
-	case q.items <- newQueueRequest(ctx, req):
+	case q.items <- queueRequest[T]{ctx: ctx, req: req}:
 		return nil
 	default:
 		return ErrQueueIsFull
 	}
 }
 
-// Poll returns a request from the queue once it's available. It returns false if the queue is stopped.
-func (q *boundedMemoryQueue[T]) Poll() (QueueRequest[T], bool) {
+// Consume applies the provided function on the head of queue. The call blocks until there is an item available.
+// It returns false if the queue is stopped.
+func (q *boundedMemoryQueue[T]) Consume(consumeFunc func(context.Context, T)) bool {
```

Comment:
return error for consumeFunc.

Reply:
What would we do with that? Quque.Consume reports a bool when the queue is stopped. My idea is that it's the responsibility of the client to handle the consumeErr before in a wrapped `consumeFunc`

Reply:
Don't you need that, to get rid of the `https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queue_sender.go#L117`?

Reply:
No. I initially tried to move the error handling inside the queue but realized that it would be a significant change in the behavior: we will always re-enqueue to the persistent queue even if the retry sender is disabled. I believe this should be configurable on the queue sender, that way, we need to keep the error handling in the queue sender. I updated https://github.com/open-telemetry/opentelemetry-collector/pull/8942 with this approach

Reply:
`we will always re-enqueue to the persistent queue even if the retry sender is disabled` only if it is a not persistent error.